boot-qemu.py

```diff
@@ -476,6 +477,35 @@ def __init__(self):
         self._initrd_arch = 'arm64be'
 
 
+class LoongArchQEMURunner(QEMURunner):
+
+    def __init__(self):
+        super().__init__()
+
+        self.cmdline.append('console=ttyS0,115200')
+        self._default_kernel_path = Path('arch/loongarch/boot/vmlinuz.efi')
+        self._initrd_arch = 'loongarch'
+
+        bios = Path(utils.BOOT_UTILS, 'images', self._initrd_arch,
+                    'edk2-loongarch64-code.fd')
+        if not bios.exists():
+            firmware_url = f"https://github.com/loongson/Firmware/raw/main/LoongArchVirtMachine/{bios.name}"
+            utils.green(
+                f"Downloading LoongArch firmware from {firmware_url}...")
+            curl_cmd = ['curl', '-LSs', '-o', bios, firmware_url]
+            subprocess.run(curl_cmd, check=True)
+
+        self._qemu_arch = 'loongarch64'
+        self._qemu_args += [
+            '-M', 'virt',
+            '-cpu', 'la464',
+            '-bios', bios,
+            '-no-reboot',
+        ]  # yapf: disable
+        self._ram = '4G'
+        self.smp = 2
```

Comment:
It could be fine to continue running with -m 2G -smp 2 (i.e. the justification in the commit message is fair) but minimally just -m 1G would suffice and take slightly less resources. (The LoongArch virt machine needs at least 1G due to memory layout requirements.)

Reply:
Good idea! I have reduced it to `2G` and I can confirm that everything still works fine.